libraries/AC_AttitudeControl/AC_AttitudeControl.cpp

```diff
@@ -238,9 +238,9 @@ void AC_AttitudeControl::input_quaternion(Quaternion& attitude_desired_quat, Vec
         // When acceleration limiting and feedforward are enabled, the sqrt controller is used to compute an euler
         // angular velocity that will cause the euler angle to smoothly stop at the input angle with limited deceleration
         // and an exponential decay specified by _input_tc at the end.
-        _ang_vel_target.x = input_shaping_angle(wrap_PI(attitude_error_angle.x), _input_tc, get_accel_roll_max_radss(), _ang_vel_target.x, ang_vel_target.x, radians(_ang_vel_roll_max), _dt);
-        _ang_vel_target.y = input_shaping_angle(wrap_PI(attitude_error_angle.y), _input_tc, get_accel_pitch_max_radss(), _ang_vel_target.y, ang_vel_target.y, radians(_ang_vel_pitch_max), _dt);
-        _ang_vel_target.z = input_shaping_angle(wrap_PI(attitude_error_angle.z), _input_tc, get_accel_yaw_max_radss(), _ang_vel_target.z, ang_vel_target.z, radians(_ang_vel_yaw_max), _dt);
+        _ang_vel_target.x = input_shaping_angle(wrap_PI(attitude_error_angle.x), _input_tc, get_accel_roll_max_radss(), _ang_vel_target.x, ang_vel_target.x, radians(_ang_vel_roll_max), get_dt());
```

Comment:
you could do:
`const float _dt = get_dt();`

Because it uses the same variable name as the old internal variable the diff would be much less. Even if its a bit to odd to reuse the variable name I think pre-fetching like that would be more efficient.

Reply:
This is where I get stupid but I figured that we were just accessing an existing and pre-calculated value. This form also represents what we are doing more accuratly I think. This is just my first go for discussion though so I am open to all and any suggestions and discussions.

Reply:
I suspect it will all get inlined anyway